parallel-install/pkg/git/clone.go

```diff
@@ -2,12 +2,28 @@ package git
 
 import (
 	"context"
+	"strings"
 
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/pkg/errors"
-	"gopkg.in/src-d/go-git.v4"
-	"gopkg.in/src-d/go-git.v4/plumbing"
 )
 
+// CloneRepo clones the repository in the given URL to the given dstPath and checks out the given revision.
+// revision can be 'main', a release version (e.g. 1.4.1), a commit hash (e.g. 34edf09a) or a PR (e.g. PR-9486).
+func CloneRepo(url, dstPath, rev string) error {
+	repo, err := defaultCloner.Clone(url, dstPath, true)
+	if err != nil {
+		return errors.Wrapf(err, "Error downloading repository (%s)", url)
+	}
+	if rev != "" {
+		return checkout(repo, url, rev)
+	}
+	return nil
+}
+
+var defaultCloner repoCloner = &remoteRepoCloner{}
```

Comment:
Please move this to the top. Package variables should always be at the top of the file.